delivery_auto_refresh/models/stock_picking.py

```diff
@@ -34,9 +34,15 @@ def _add_delivery_cost_to_so(self):
             volume += (move_line.product_id.volume or 0.0) * qty
             quantity += qty
             total += move.sale_line_id.price_unit * qty
-        total = sale_order.currency_id._convert(
-            total, sale_order.company_id.currency_id, sale_order.company_id,
-            sale_order.date_order)
+        # In v11 there's no _convert method to get proper currency conversion
+        # according to a given date. We replicate such behavior here.
+        if sale_order.company_id.currency_id != sale_order.currency_id:
+            to_currency = sale_order.company_id.currency_id
+            from_currency = sale_order.currency_id
+            total = total * from_currency._get_conversion_rate(
+                from_currency, to_currency,
+                sale_order.company_id, sale_order.date_order)
+            total = to_currency.round(total)
```

Comment:
You have to use instead the method `compute`:
```python
total = sale_order.currency_id.with_context(date=sale_order.date_order).compute(total, sale_order.company_id.currency_id)
```

Reply:
:+1: